File: 2022/aoc_2022/test/day_3_test.exs
defmodule Day3Test do
  use ExUnit.Case
  doctest Aoc2022.Day3

  test "part 1-example" do
    assert Aoc2022.Day3.solve("assets/day3_example.txt") == 157
  end

  test "part 1" do
    assert Aoc2022.Day3.solve("assets/day3.txt") == 8240
  end


  test "part 2-example" do
    assert Aoc2022.Day3.solve2("assets/day3_example.txt") == 70
  end

  test "part 2" do
    assert Aoc2022.Day3.solve2("assets/day3.txt") == 2587
  end

end

File: 2024/aoc2024.Tests/Day3Test.cs
using System;
using System.IO;
using aoc2024;
using JetBrains.Annotations;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace aoc2024.Tests;

[TestClass]
[TestSubject(typeof(Day3))]
public class Day3Test
{

    [TestMethod]
    public void testSolve1Example()
    {
        var projectDirectory = Directory.GetParent(AppContext.BaseDirectory).FullName;
        var filePath = Path.Combine(projectDirectory, "inputs", "day3-example.txt");
        Assert.AreEqual(161, Day3.solve1(filePath));
    }
    [TestMethod]
    public void testSolve1()
    {
        var projectDirectory = Directory.GetParent(AppContext.BaseDirectory).FullName;
        var filePath = Path.Combine(projectDirectory, "inputs", "day3-input.txt");
        Assert.AreEqual(168539636, Day3.solve1(filePath));
    }
    
    [TestMethod]
    public void testSolve2Example()
    {
        var projectDirectory = Directory.GetParent(AppContext.BaseDirectory).FullName;
        var filePath = Path.Combine(projectDirectory, "inputs", "day3-2-example.txt");
        Assert.AreEqual(48, Day3.solve2(filePath));
    }
    [TestMethod]
    public void testSolve2()
    {
        var projectDirectory = Directory.GetParent(AppContext.BaseDirectory).FullName;
        var filePath = Path.Combine(projectDirectory, "inputs", "day3-input.txt");
        Assert.AreEqual(97529391, Day3.solve2(filePath));
    }
}